code/game/machinery/chemHandC.dm

```diff
@@ -81,77 +96,86 @@
 	else
 		return ..()
 
-/obj/machinery/chemheater/attack_ghost()
+/obj/machinery/chemtemper/attack_ghost()
 	return
 
-/obj/machinery/chemheater/attack_hand(mob/user)
-	if(held_container)
-		overlays -= onstage
-		to_chat(user, "<span class='notice'>You remove \the [held_container] from \the [src].</span>")
-		user.put_in_hands(held_container)
-		held_container = null
-		had_item = TRUE
-	toggle()
-	had_item = FALSE
+/obj/machinery/chemtemper/AltClick(mob/user)
+	if(!user.incapacitated() && Adjacent(user) && !(stat & (FORCEDISABLE|NOPOWER) && user.dexterity_check()))
```

Comment:
Parenthesis issue - `!(stat & (FORCEDISABLE|NOPOWER) && user.dexterity_check())` will be TRUE if the machine lacks power and the user passes the dexterity check or the inverse. Probably should be the following instead: `!(stat & (FORCEDISABLE|NOPOWER)) && user.dexterity_check()`